lib/logstash/plugin_mixins/elasticsearch/common.rb

```diff
@@ -206,19 +219,23 @@ def next_sleep_interval(current_interval)
       doubled > @retry_max_interval ? @retry_max_interval : doubled
     end
 
-    def handle_dlq_status(message, action, status, response)
+    def handle_dlq_response(message, action, status, response)
+      _, action_params = action.event, [action[0], action[1], action[2]]
+
+      # TODO: Change this to send a map with { :status => status, :action => action } in the future
+      detailed_message = "#{message} status: #{status}, action: #{action_params}, response: #{response}"
+
+      log_level = dig_value(response, 'index', 'error', 'type') == 'invalid_index_name_exception' ? :error : :warn
+
+      handle_dlq_status(action, log_level, detailed_message)
+    end
+
+    def handle_dlq_status(action, log_level, message)
       # To support bwc, we check if DLQ exists. otherwise we log and drop event (previous behavior)
       if @dlq_writer
-        event, action = action.event, [action[0], action[1], action[2]]
-        # TODO: Change this to send a map with { :status => status, :action => action } in the future
-        @dlq_writer.write(event, "#{message} status: #{status}, action: #{action}, response: #{response}")
+        @dlq_writer.write(action.event, "#{message}")
       else
-        if dig_value(response, 'index', 'error', 'type') == 'invalid_index_name_exception'
-          level = :error
-        else
-          level = :warn
-        end
-        @logger.send level, message, status: status, action: action, response: response
```

Comment:
@andsel Now we can globally use `handle_dlq_status` without relying on `response` (3-rd party data).
And this particular line is changed from map to ordinary string in order to standardize, rest of structures and logics are same. Any ideas if this bring issue?

Reply:
I don't foresee issues, it doesn't instantiate more new objects than before, so the pressure in GC should remain the same.
This new version of `handle_dlq_status` could be easily reused also in PS #1084 (https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1084/files#diff-940291b716938425b715566d94fceb6a35ac05e51f901276637846a9abc77737R361-R365)

Reply:
Thanks for the confirmation!